### ISSUE TITLE
Hello, I would like to ask why there is such a difference between Python and C++.

### ISSUE THREAD

**Issue by zhaochenxiao90** [catboost/catboost#2739]
Problem:
python :
model = CatBoostClassifier()
model.load_model("simple_catboost_model.cbm")
print(model.predict_proba([[1, 0, 2]]))

output:
[[0.76991639 0.23008361]]
C++ :
#include <wrapped_calcer.h>
#include <iostream>

int main() {
    try {
        // 使用 ModelCalcerWrapper 加载 CatBoost 模型
        ModelCalcerWrapper calcer("20240909150255_catboost_model.cbm");

        // 获取使用的特征名称
        auto featureNames = calcer.GetUsedFeaturesNames();

        // 输出特征名称
        std::cout << "Used features names:" << std::endl;
        for (const auto& name : featureNames) {
            std::cout << name << std::endl;
        }

        // 定义浮点特征和分类特征
        std::vector<float> floatFeatures = {1.0, 0.0, 2.0};
        // 传递空的分类特征
        std::vector<std::string> catFeatures; // 空的分类特征

        // 进行预测
        double prediction = calcer.Calc(floatFeatures, catFeatures);

        // 输出预测结果
        std::cout << "Prediction: " << prediction << std::endl;
    } catch (const std::exception &e) {
        std::cerr << "Error: " << e.what() << std::endl;
        return 1;
    }

    return 0;
}
output:
Prediction: -1.20784
Why is that? Is there anything wrong with me?

**Comment by zhaochenxiao90** [comment]
X = [
    [1, 0, 2],
    [2, 1, 3],
    [3, 0, 1],
    [4, 1, 2],
    [5, 0, 3],
    [6, 1, 1]
]
y = [0, 1, 0, 1, 0, 1]

# 将数据集分成训练集和测试集
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

# 创建 CatBoostClassifier 模型
model = CatBoostClassifier(iterations=100, learning_rate=0.1, depth=3, verbose=0)

# 训练模型
model.fit(X_train, y_train)

# 预测
y_pred = model.predict(X_test)
y_pred_proba = model.predict_proba(X_test)

# 输出预测结果
print(f"Predicted classes: {y_pred}")
print(f"Predicted probabilities: {y_pred_proba}")

# 计算准确率
accuracy = accuracy_score(y_test, y_pred)
print(f"Accuracy: {accuracy}")

# 保存模型
model.save_model("simple_catboost_model.cbm")

This is the training and preservation of the model

**Comment by andrey-khropov** [comment]
> Why is that?

That's because they output different values.
C/C++ applier's Calc function outputs raw model prediction as stated in [the documentation](https://github.com/catboost/catboost/blob/335c6cf9d2a9fc2df56922f7eaa78bcaa697f06c/catboost/libs/model_interface/wrapped_calcer.h#L120).

But when you use `predict_proba` for binary classification it returns the probability of negative and positive classes.
They are computed from raw formula output like that:

P_positive = sigmoid(raw_model_prediction)
P_negative = 1 - P_positive.

You can easily verify that:

```
$  python -c "import math; raw_formula_val = -1.20784; p_pos = 1.0 / (1.0 + math.exp(-raw_formula_val)); p_neg = 1 - p_pos; print (p_neg, p_pos) "
0.7699165388139099 0.23008346118609016
```

We'll improve documentation about this.

**Comment by zhaochenxiao90** [comment]
Thank you, this is truly a magical experience
